lib/eth/abi.rb

```diff
@@ -381,6 +410,39 @@ def encode_bytes(arg, type)
       end
     end
 
+    # Properly encodes tuples.
+    def encode_tuple(arg, type)
+      raise EncodingError, "Expecting Hash: #{arg}" unless arg.instance_of? Hash
+      raise EncodingError, "Expecting #{type.components.size} elements: #{arg}" unless arg.size == type.components.size
+
+      static_size = 0
+      type.components.size.times do |i|
```

Comment:
If you are looping through components, it's a lot more convenient, readable, and idiomatic to use `each` instead of index-based looping.

```suggestion
      type.components.each do |component|
```

If you are looping through a collection and updating a value with each iteration, you can use `inject`/`reduce` or `each_with_object`.

```ruby
static_size = type.components.inject(0) do |component, total|
  if component.is_dynamic?
    32
  else
    Util.ceil32(component.size || 0)
  end
end
```

If you want to get the sum of a collection, you can use `sum`. Simple example:

```ruby
[1, 2, 3].sum => 6
```

And if it has conditions, you can pass `sum` a block.

```ruby
static_size = type.components.sum { |c| c.is_dynamic? ? 32 : Util.ceil32(c.size || 0) }
```